OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -77,7 +77,8 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 					var teamHeader = ScrollItemWidget.Setup(teamTemplate, () => true, () => { });
 					teamHeader.Get<LabelWidget>("TEAM").GetText = () => t.Key == 0 ? "No Team" : "Team {0}".F(t.Key);
 					var teamRating = teamHeader.Get<LabelWidget>("TEAM_SCORE");
-					teamRating.GetText = () => t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString();
+					var scoreCache = new CachedTransform<int, string>(s => s.ToString());
+					teamRating.GetText = () => scoreCache.Update(t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0));
```

Comment:
Can win a bit more here by caching and reusing the score enumerable:
```
var teamMemberScores = t.Select(tt => tt.Second)
	.Where(s => s != null)
	.ToArray()
	.Select(s => s.Experience);

	teamRating.GetText = () => scoreCache.Update(teamMemberScores.Sum());
```

Reply:
done